run_pipeline.py

```diff
@@ -1,63 +1,107 @@
 import logging
 import os
-from config import s3_endpoint_url, s3_bucket, s3_folder_in_bucket
+import time
+from typing import Optional
+from config import (
+    data_base_dir,
+    s3_endpoint_url,
+    ae_samplerate_hz,
+    ae_file_extension,
+    ae_convert_to_mono,
+)
 from download import download_uri
-from base_util import get_asset_info
-from s3_util import S3Store
-from transcode import ffmpeg_transcode
+from base_util import (
+    get_asset_info,
+    remove_all_input_output,
+    save_provenance,
+    PROVENANCE_JSON_FILE,
+)
+from s3_util import S3Store, parse_s3_uri
+from transcode import ffmpeg_audio_extraction
 
 logger = logging.getLogger(__name__)
 
 
-def run(input_uri: str, output_uri: str) -> bool:
+def run(input_uri: str, output_uri: str) -> Optional[str]:
     logger.info(f"Processing {input_uri} (save to --> {output_uri})")
```

Comment:
Perhaps do an early check to see whether the output_uri exists already (for instance, if the staged asset is already an mp3 file). Might need to check that the object corresponding to the output_uri fulfills some criteria (sample rate, etc.)

Reply:
1. `output_uri` is a path to a directory, not a file. The string of `output_uri` will be concatenated with the resulting audio file from the worker, and then that entire URI will be used to upload the output to S3.
2. Let's say there is an audio file already in that directory. How will I look for it AND also find out info about sample rate, etc? I think this would add additional unnecessary logic to the worker. We assume that there is no audio present in `output_uri` and, if there is, it will be overwritten. The only check that makes sense is for the input to not be audio already. Maybe I can also go beyond that and try to download it, then extract the `sample_rate` and whether it's mono audio or not, then apply ffmpeg if those parameters don't match as well.

Reply:
Like we discussed in the office: 
Airflow should be responsible for checking whether the audio file exists already, the worker should not need to bother.
I'm not sure it's relevant to check sample rate/ channels at this point, if I understood correctly Whisper will make sure of this. So I guess we can leave this for now.